### PR TITLE
[Application logger] Correct log file object path if PIMCORE_PROJECT_ROOT is /app

### PR DESCRIPTION
**Pull request by BlackbitDevs**
Follow-up of #11555 where we forgot this place.

### CODE REVIEW

File: lib/Log/ApplicationLogger.php [hunk 1]
```diff
@@ -150,9 +150,9 @@ public function log($level, $message, array $context = [])
 
         if (isset($context['fileObject'])) {
             if (is_string($context['fileObject'])) {
-                $context['fileObject'] = str_replace(PIMCORE_PROJECT_ROOT, '', $context['fileObject']);
+                $context['fileObject'] = preg_replace('/^'.preg_quote(\PIMCORE_PROJECT_ROOT, '/').'/', '', $context['fileObject']);
             } else {
```
Comment (brusch) [inline lib/Log/ApplicationLogger.php]: Should we maybe add an `instanceof` here?

**Review by brusch — CHANGES_REQUESTED** [review]

File: lib/Log/ApplicationLogger.php (hunk 1 above)
Comment (BlackbitDevs) [reply lib/Log/ApplicationLogger.php]: In this `is_string` case? Or do you mean https://github.com/pimcore/pimcore/pull/11601/files#diff-935282905cd71faeda62809e162086e37f92682e8f041d8e6e40eafe55f2c11cR155

Comment (brusch) [reply lib/Log/ApplicationLogger.php]: Yeah, I mean changing the `else` to an `elseif($context['fileObject'] instanceof FileObject)`.

Comment (BlackbitDevs) [reply lib/Log/ApplicationLogger.php]: Fixed

**Comment by brusch** [comment]
Nice, thanks a lot! 👍

**Pull request merged**